### DISCUSSION TITLE
Please add a place to annotate the kube-system namespace

### DISCUSSION THREAD

**Discussion by BrianMusson** [gruntwork-io/knowledge-base#336] [Help] [answered]
I need to add annotations and labels to the pre-existing kube-system namespace. I looked at using the k8s-namespace module and the k8s core services modules but did not see what I was looking for. In the route53 module, you did the "created_outside_of_terraform" bool. I think that would work well as an addition to the k8s-namespace module. Another idea would be to add namespace annotations and labels to the core services module so that it could apply them to cluster with the local-exec provisioner.

related: https://github.com/hashicorp/terraform-provider-kubernetes/issues/692

[r:terraform-aws-service-catalog](https://github.com/gruntwork-io/terraform-aws-service-catalog)
[r:terraform-aws-eks](https://github.com/gruntwork-io/terraform-aws-eks)
Thanks!

**Comment by BrianMusson** [comment, marked as answer]
I was able to work around this by using a new feature from the v2.10 EKS provider

https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs/resources/labels
https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs/resources/annotations
```
resource "kubernetes_annotations" "kube-system" {
  depends_on = [
    helm_release.linkerd2,
    helm_release.linkerd2-cni
  ]

  force       = true
  api_version = "v1"
  kind        = "Namespace"
  metadata {
    name = "kube-system"
  }
  annotations = {
    "linkerd.io/inject" = "disabled"
  }
}

resource "kubernetes_labels" "kube-system" {
  depends_on = [
    helm_release.linkerd2,
    helm_release.linkerd2-cni
  ]

  force       = true
  api_version = "v1"
  kind        = "Namespace"
  metadata {
    name = "kube-system"
  }
  labels = {
    "config.linkerd.io/admission-webhooks" = "disabled"
  }
}
```

**Reply by rhoboat** [reply]
Thanks for reporting back. I've opened an [issue](https://github.com/gruntwork-io/terraform-aws-eks/issues/427) to track and plan support for annotations and labels.
